### PR TITLE
refactor(mojaloop/#2802): rework party lookup loop and count increment logic

### PR DESCRIPTION
**Pull request by kleyow**
refactor(mojaloop/#2802): rework party lookup loop and count increment logic - https://github.com/mojaloop/project/issues/2802

- Update sequence diagram for command handler breaking party loop
- Rework handlers for command handler breaking party loop
- Remove handlers from domain handler that are no longer needed
- Fix increment logic that was leading to duplicate event messages in party lookup
- Fix increment logic to mitigate duplicate event messages in bulk quotes

### CODE REVIEW

File: modules/private-shared-lib/src/infra/redis_bulk_transaction_repo.ts [hunk 1]
```diff
@@ -327,13 +327,19 @@ export class RedisBulkTransactionStateRepo implements IBulkTransactionEntityRepo
...
 
-    async incrementBulkQuotesSuccessCount(bulkId: string): Promise<void> {
+    async incrementBulkQuotesSuccessCount(bulkId: string, increment = 1): Promise<number> {
         if(!this.canCall()) {
             throw (new Error('Repository not ready'));
         }
         const key: string = this.keyWithPrefix(bulkId);
         try {
-            await this._redisClient.hIncrBy(key, this.bulkQuotesSuccessCountKey, 1);
+            const newValue = await this._redisClient.hIncrBy(key, this.bulkQuotesSuccessCountKey, 1);
```
Comment (vijayg10) [inline modules/private-shared-lib/src/infra/redis_bulk_transaction_repo.ts:336]:
```suggestion
            const newValue = await this._redisClient.hIncrBy(key, this.bulkQuotesSuccessCountKey, increment);
```

File: modules/private-shared-lib/src/infra/redis_bulk_transaction_repo.ts [hunk 2]
```diff
@@ -375,13 +381,19 @@ export class RedisBulkTransactionStateRepo implements IBulkTransactionEntityRepo
...
 
-    async incrementBulkQuotesFailedCount(bulkId: string): Promise<void> {
+    async incrementBulkQuotesFailedCount(bulkId: string, increment = 1): Promise<number> {
         if(!this.canCall()) {
             throw (new Error('Repository not ready'));
         }
         const key: string = this.keyWithPrefix(bulkId);
         try {
-            await this._redisClient.hIncrBy(key, this.bulkQuotesFailedCountKey, 1);
+            const newValue = await this._redisClient.hIncrBy(key, this.bulkQuotesFailedCountKey, 1);
```
Comment (vijayg10) [inline modules/private-shared-lib/src/infra/redis_bulk_transaction_repo.ts:390]:
```suggestion
            const newValue = await this._redisClient.hIncrBy(key, this.bulkQuotesFailedCountKey, increment);
```

**Review by vijayg10 — CHANGES_REQUESTED** [review]

File: modules/private-shared-lib/src/infra/redis_bulk_transaction_repo.ts (hunk 1 above)
Comment (kleyow) [reply modules/private-shared-lib/src/infra/redis_bulk_transaction_repo.ts:336]: good catch, thanks. done.

**Review by mdebarros — APPROVED** [review]
+1

@kleyow we also need to get this change into the main mvp branch and the transfers feature branch.

**Pull request merged**